smartsim/_core/entrypoints/dragon_install.py

```diff
@@ -0,0 +1,309 @@
+    asset = filter_assets(asset_map)
+    if asset is None:
+        raise SmartSimCLIActionCancelled("No dragon runtime asset available to install")
+
+    logger.debug(f"Retrieved asset metadata: {asset}")
+    return asset
+
+
+def retrieve_asset(working_dir: pathlib.Path, asset: GitReleaseAsset) -> pathlib.Path:
+    """Retrieve the physical file associated to a given GitHub release asset
+
+    :param asset: GitHub release asset to retrieve
+    :type asset: GitReleaseAsset
+    :returns: path to the downloaded asset
+    :rtype: pathlib.Path"""
+    output_path = working_dir / asset.name
+    if output_path.exists():
+        return output_path
+
+    request = requests.get(asset.browser_download_url, timeout=60)
+    status_code = request.status_code
+
+    if status_code != 200:
+        raise SmartSimCLIActionCancelled(
+            f"Unable to retrieve asset. Request status {status_code}"
+        )
+
+    with open(output_path, "wb") as asset_file:
+        asset_file.write(request.content)
+
+    logger.debug("Selected asset: {filename}")
+    return output_path
+
+
+def expand_archive(archive_path: pathlib.Path) -> pathlib.Path:
+    """Expand the archive file from the asset
+
+    :param archive_path: path to a downloaded archive for a release asset
+    :type archive_path: Optional[pathlib.Path]"""
+    if not archive_path.exists():
+        raise ValueError(f"Archive {archive_path} does not exist")
+
+    # create a target dir that excludes the .tar.gz suffixes
+    expand_to = str(archive_path.absolute()).replace(".tar.gz", "")
+
+    with tarfile.TarFile.open(archive_path, "r") as archive:
+        archive.extractall(expand_to)
+
+    logger.debug("Asset expanded into: {expand_to}")
+    return pathlib.Path(expand_to)
+
+
+def install_package(asset_dir: pathlib.Path) -> int:
+    """Install the package found in `asset_dir` into the current python environment
+
+    :param asset_dir: path to a decompressed archive contents for a release asset
+    :type asset_dir: Optional[pathlib.Path]"""
+    package_path = next(asset_dir.rglob("*.whl"), None)
+    if not package_path:
+        logger.error(f"No wheel found for package in {asset_dir}")
+        return 1
+
+    cmd = f"python -m pip install --force-reinstall {package_path}"
+    logger.info(f"Executing installation: {cmd}")
+
+    with subprocess.Popen(cmd.split()) as installer:
+        result = installer.wait()
+        logger.debug(f"Installation completed with return code: {result}")
+        return result
+
+
+def cleanup(
+    archive_path: t.Optional[pathlib.Path] = None,
+    asset_dir: t.Optional[pathlib.Path] = None,
+) -> None:
+    """Delete the downloaded asset and any files extracted during installation
+
+    :param archive_path: path to a downloaded archive for a release asset
+    :type archive_path: Optional[pathlib.Path]
+    :param asset_dir: path to a decompressed archive contents for a release asset
+    :type asset_dir: Optional[pathlib.Path]
```

Comment:
As we're moving away from docstring `:type:` annotations in `develop`, feel free to remove them already.

Reply:
Gladly!

Reply:
Removed all `:type...` and `:rtype:` from docstrings